.github/workflows/tf_apply_production.yml

```diff
@@ -0,0 +1,44 @@
+name: "Terraform Apply Production"
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "terragrunt/**"
+      - ".github/workflows/tf_apply_production.yml"
+
+env:
+  AWS_REGION: ca-central-1
+  TERRAFORM_VERSION: 1.3.9
+  TERRAGRUNT_VERSION: 0.38.4
+  CONFTEST_VERSION: 0.27.0
+
+permissions:
+  id-token: write
+  contents: read
+  pull-requests: write
+  actions: write
+  checks: write
+  statuses: write
+  
+job:
+  terragrunt-apply:
+  runs-on: ubuntu-latest
+  steps:
+    - name: Checkout
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
+
+    - name: setup terraform tools
+      uses: cds-snc/terraform-tools-setup@v1
+
+    - name: configure aws credentials using OIDC
+      uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # tag=v1.7.0
+      with:
+        role-to-assume: arn:aws:iam::414662622316:role/simplify-privacy-statements-V2-apply
+        role-session-name: Terraform Apply
+        aws-region: ${{ env.AWS_REGION }}
+
+    - name: Terraform Apply
+      working-directory: terragrunt/env/production/
+      run: terragrunt apply --terragrunt-non-interactive -auto-approve           
```

Comment:
One last thing I just noticed - this needs to be indented one level below the `terragrunt-apply` key.  The plan job will need the same fix.